src/openmc_heat_driver.cpp

```diff
@@ -196,38 +196,29 @@ void OpenmcHeatDriver::init_tallies()
 void OpenmcHeatDriver::init_densities()
 {
   if (this->has_global_coupling_data()) {
-    std::size_t n_solid =
-      heat_driver_->n_pins_ * heat_driver_->n_axial_ * heat_driver_->n_rings();
     std::size_t n_fluid = heat_driver_->n_pins_ * heat_driver_->n_axial_;
-    densities_.resize({n_solid + n_fluid});
-    densities_prev_.resize({n_solid + n_fluid});
+    densities_.resize({n_fluid});
+    densities_prev_.resize({n_fluid});
 
     if (density_ic_ == Initial::neutronics) {
       // Loop over all of the fluid regions in the heat transfer model and set
-      // the density IC based on the densities used in the OpenMC input file. Set
-      // the initial solid density to 0.0 because it is not used in the simulation
-      // at all anyways.
-      densities_.fill(0.0);
+      // the density IC based on the densities used in the OpenMC input file.
 
       int fluid_index = 0;
-      int fluid_offset =
-        heat_driver_->n_pins_ * heat_driver_->n_axial_ * heat_driver_->n_rings();
-      for (gsl::index i = 0; i < heat_driver_->n_pins_; ++i) {
-        for (gsl::index j = 0; j < heat_driver_->n_axial_; ++j) {
-
-          double mass = 0.0;
-          double total_vol = 0.0;
-
-          for (const auto& idx : elem_to_cell_inst_[fluid_index++]) {
-            const auto& c = openmc_driver_->cells_[idx];
-            double vol = c.volume_;
+      int n_fluid_elems = heat_driver_->n_pins_ * heat_driver_->n_axial_;
+      for (gsl::index fluid_index = 0; fluid_index < n_fluid_elems; ++fluid_index) {
```

Comment:
Looks like you don't need to declare `n_fluid_elems`.  I think you can use `n_fluid`, which was declared above.